docs/maintaining.md

```diff
@@ -92,3 +92,118 @@ made to always install the latest released version.
+`release-vX.Y.Z-grafana`). The release branches will always use the same release
+version as the `prometheus/prometheus` release it is based off of.
+
+By adding features to the `grafana/prometheus` repository first, we are
+committing ourselves to extra maintenance of features that have not yet been
+merged upstream. Feature authors will have to babysit their features to
+coordinate with the Prometheus release schedule to always be compatible. One the
+feature is merged upstream, this burden of maintaining is made easier as the
+Prometheus team can more easily sync breaking changes across the whole
+repository.
+
+We are purposefully carrying this extra burden because we intend to ultimately
+make Prometheus better and contribute all of our enhancements upstream. We want
+to strive to benefit the Prometheus ecosystem at large.
+
+### Creating a New Feature
+
+For `grafana/prometheus` maintainers to create a new feature, they will do the
+following:
+
+1. Create a feature branch in `grafana/prometheus` based on the latest release
+   tag that `grafana/prometheus` currently has a release branch for. The feature
+   branch should follow the naming convention `feat-<feature name>`.
+2. Implement the feature and open a PR to merge the feature branch into the
+   associated `grafana/prometheus` release branch.
+3. Once the release branch is updated, open a PR to update `grafana/agent` to
+   use the latest release branch SHA.
+
+### Updating an Existing Feature
+
+If a feature branch that was already merged to a release branch needs to be
+updated for any reason:
+
+1. Push directly to the feature branch or open a PR to merge changes into that
+   feature branch.
+2. Open a PR to merge the new changes from the feature branch into the
+   associated release branch.
+3. Once the release branch is updated, open a PR to update `grafana/agent` to
+   use the latest release branch SHA.
+
+### Handling New Upstream Release
+
+When a new upstream `prometheus/prometheus` release is available, we must go
+through the following process:
+
+1. Create a new `grafana/prometheus` release branch named
+   `release-X.Y.Z-grafana`.
+2. For all feature branches still not merged upstream, rebase them on top of the
+   latest release. Force push them to update the `grafana/prometheus` release
+   branch.
+3. Create one or more PRs to introduce the features into the newly created
+   release branch.
+
+Once this process is completed, the previous release branch in
+`grafana/prometheus` is considered stale and will no longer be updated.
+
+### Updating the Agent's vendor
+
+The easiest way to do this is the following:
+
+1. Edit `go.mod` and change the replace directive to the release branch name.
+2. Run `go mod vendor && go mod tidy`.
+3. Commit and open a PR.
+
+### Gotchas
+
+If the `grafana/prometheus` feature is incompatible with the upstream master
+branch, an upstream PR cannot be merged due to the merge conflicts. There are a
```

Comment:
I'm not following here, we're using the word upstream twice in a different context - am I missing something?